src/core/scalar.rs

```diff
@@ -0,0 +1,312 @@
+                )
+            }
+            ScalarType::Int64 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 64-bit signed integer.",
+                )
+            }
+            ScalarType::Int128 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 128-bit signed integer.",
+                )
+            }
+            ScalarType::UInt8 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is an 8-bit unsigned integer.",
+                )
+            }
+            ScalarType::UInt16 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 16-bit unsigned integer.",
+                )
+            }
+            ScalarType::UInt32 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 32-bit unsigned integer.",
+                )
+            }
+            ScalarType::UInt64 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 64-bit unsigned integer.",
+                )
+            }
+            ScalarType::UInt128 => {
+                self.schema_inner(
+                    InstanceType::Integer,
+                    "A field whose value is a 128-bit unsigned integer.",
+                )
+            }
+            ScalarType::Bytes => {
+                self.schema_inner(
+                    InstanceType::String,
+                    "A field whose value is a sequence of bytes.",
+                )
+            }
+        }
+    }
+}
+
+lazy_static! {
+    pub static ref CUSTOM_SCALARS: HashMap<String, ScalarType> = {
+        let scalars: Vec<ScalarType> = vec![
+            ScalarType::Empty,
+            ScalarType::Email,
+            ScalarType::PhoneNumber,
+            ScalarType::Date,
+            ScalarType::Url,
+            ScalarType::JSON,
+            ScalarType::Int8,
+            ScalarType::Int16,
+            ScalarType::Int32,
+            ScalarType::Int64,
+            ScalarType::Int128,
+            ScalarType::UInt8,
+            ScalarType::UInt16,
+            ScalarType::UInt32,
+            ScalarType::UInt64,
+            ScalarType::UInt128,
+            ScalarType::Bytes,
+        ];
+        let mut hm = HashMap::new();
+
+        for scalar in scalars {
+            hm.insert(scalar.name(), scalar);
+        }
+        hm
+    };
+}
+lazy_static! {
+    static ref SCALAR_TYPES: HashSet<&'static str> = {
+        let mut set = HashSet::new();
+        set.extend(["String", "Int", "Float", "ID", "Boolean"]);
+        set.extend(CUSTOM_SCALARS.keys().map(|k| k.as_str()));
+        set
+    };
```

Comment:
Move to default implementation of Scalar

Reply:
??